### PR TITLE
Db data set

### PR DESCRIPTION
**Pull request by JorisGoosen**
Implements https://github.com/jasp-stats/INTERNAL-jasp/issues/1194

### CODE REVIEW

**Comment by JorisGoosen** [comment]
So reviews would be nice ^^

**Comment by JorisGoosen** [comment]
@RensDofferhoff if I load debugdataset and check whether the missing values are correct:
image

So im not sure why it isnt working with your compile?
Did you manually make the data as opposed to loading it maybe?

**Comment by JorisGoosen** [comment]
Ah yes 
image

**Comment by JorisGoosen** [comment]
Fixed it
afbeelding

Problem was that SQLITE couldnt properly store nan/inf

File: Tools/CMake/Modules.cmake
Comment (boutinb) [inline Tools/CMake/Modules.cmake:1]: Set right alphabetical order

File: QMLComponents/controls/jaspcontrol.h [hunk 1]
```diff
@@ -10,7 +10,7 @@
 class AnalysisForm;
 class JASPListControl;
 class BoundControl;
-class ComputedColumn;
+class Column;
```
Comment (boutinb) [inline QMLComponents/controls/jaspcontrol.h]: Forward def not needed anymore apparently

File: QMLComponents/components/JASP/Theme/Theme.qml [hunk 2]
```diff
@@ -61,7 +61,7 @@ JaspTheme
 	fontRCode.weight:									Font.Normal
 
 	fontCode.family:									preferencesModel.codeFont
-	fontCode.pixelSize:									Math.round(10 * uiScale)
+	fontCode.pixelSize:									Math.round(11 * uiScale)
```
Comment (boutinb) [inline QMLComponents/components/JASP/Theme/Theme.qml:64]: Why?

Comment (JorisGoosen) [reply QMLComponents/components/JASP/Theme/Theme.qml:64]: Because it was supertiny

File: Tools/CMake/Modules.cmake [hunk 3]
```diff
@@ -19,38 +19,39 @@
...
+  "jaspLearnBayes"
+  "jaspLearnStats"
+  "jaspMachineLearning"
+  "jaspMetaAnalysis"
+  "jaspNetwork"
+  "jaspQualityControl"
+  "jaspPower"
+  "jaspPredictiveAnalytics"
+  "jaspProphet"
+  "jaspReliability"
```
Comment (JorisGoosen) [inline Tools/CMake/Modules.cmake]:
```suggestion
  "jaspPower"
  "jaspPredictiveAnalytics"
  "jaspProphet"
  "jaspQualityControl"
  "jaspReliability"
```

**Comment by boutinb** [comment]
Add row on the last row, or add column after the last column or before the first column crash

**Comment by boutinb** [comment]
I can open a file of 250 MB without problem (it is quicker than in 0.17.1), but closing the dataset is really long. I think the dbDelete function can be more efficient.

**Comment by JorisGoosen** [comment]
> I can open a file of 250 MB without problem (it is quicker than in 0.17.1), but closing the dataset is really long. I think the dbDelete function can be more efficient.

Good point, I didnt really look into this (just killed JASP every time so didnt really notice :p)

**Comment by boutinb** [comment]
This branch could not be compiled anymore. I've just fixed it ([Suppress _info redefinition in columnsModel.cpp](https://github.com/jasp-stats/jasp-desktop/pull/5020/commits/16da84b813a208e82760423f677374a1622f7512), but now JASP crashes when it starts...

**Comment by JorisGoosen** [comment]
Ah you should not have removed the definition in the cpp.
Instead you should have made it static again in ColumnsModel.h

**Comment by boutinb** [comment]
> Ah you should not have removed the definition in the cpp.
> Instead you should have made it static again in ColumnsModel.h

I set it as it is in stable.

**Comment by JorisGoosen** [comment]
Ah exporting to csv also seems to have a small problem.
And the crash on insert is only for loaded datasets

**Comment by boutinb** [comment]
I can now add columns on an existing datafile.
But adding a row still deletes values of some columns

**Comment by JorisGoosen** [comment]
New commit should fix that ^^

Ill have a look at dbDelete

**Comment by JorisGoosen** [comment]
Reminder to myself: be sure to check loading from an external database

**Comment by boutinb** [comment]
Work better, but deleting a row remove data in some columns.

**Comment by JorisGoosen** [comment]
Haha, well then I know what I will start with tomorrow :p

**Comment by JorisGoosen** [comment]
Hmm, you cant actually click a cell twice for editing, you first need to click something else...

File: Desktop/qquick/datasetview.cpp [hunk 4]
```diff
@@ -884,39 +903,59 @@ void DataSetView::updateExtraColumnItem()
...
+	
+	if(createItem && !(_prevEditRow == -1 || _prevEditCol == -1))
+	{
+		Log::log() << "Restoring text item for old edit item at " << _prevEditRow << ", " << _prevEditCol << std::endl;
+		QQuickItem * item = createTextItem(_prevEditRow, _prevEditCol);
+
+		size_t col=_prevEditCol, row=_prevEditRow;
+
+		//A delay might help the focus problem? No it doesnt...
+		QTimer::singleShot(10, _cellTextItems[col][row]->item, [col, row, this](){ _cellTextItems[col][row]->item->forceActiveFocus(); });
```
Comment (boutinb) [inline Desktop/qquick/datasetview.cpp]:
```suggestion
		QTimer::singleShot(10, _cellTextItems[col][row]->item, [col, row, this](){ if (_cellTextItems.contains(col) && _cellTextItems[col].contains(row) && _cellTextItems[col][row]->item) _cellTextItems[col][row]->item->forceActiveFocus(); });
```

**Comment by boutinb** [comment]
Let's merge it!

**Review by boutinb — APPROVED** [review]

**Pull request merged**